cl_sii/dte/data_models.py

```diff
@@ -466,7 +468,174 @@ def as_dte_data_l1(self) -> DteDataL1:
+
+    DTE doc XML element: '..//Documento//Referencia//RUTOtr'
+
+    .. note::
+        > Sólo si el documento de referencia es de tipo tributario y fue emitido
+        > por otro contribuyente
+    """
+
+    codigo_ref: Optional[CodigoReferencia] = None
+    """
+    The type of use for the reference
+
+    > Tipo de Uso de la Referencia
+
+    DTE doc XML element: '..//Documento//Referencia//CodRef'
+    """
+
+    razon_ref: Optional[str] = None
+    """
+    The reason the document is being referenced
+
+    > Razon Explicita por la que se Referencia el Documento
+
+    DTE doc XML element: '..//Documento//Referencia//RazonRef'
+    """
+
+    def __post_init__(self) -> None:
+        """
+        Run validation automatically after setting the fields values.
+
+        :raises TypeError, ValueError:
+
+        """
+
+        if (
+            self.numero_linea_ref < constants.DTE_REFERENCIA_LINE_NUMBER_MIN_VALUE
+            or self.numero_linea_ref > constants.DTE_REFERENCIA_LINE_NUMBER_MAX_VALUE
+        ):
+            raise ValueError(
+                "Value 'numero_linea_ref' must be a value between "
+                f"{constants.DTE_REFERENCIA_LINE_NUMBER_MIN_VALUE} and "
+                f"{constants.DTE_REFERENCIA_LINE_NUMBER_MAX_VALUE}",
+                self.numero_linea_ref)
+
+        if len(self.tipo_documento_ref) < 1 or len(self.tipo_documento_ref) > 3:
+            raise ValueError(
+                "The length of 'tipo_documento_ref' must be a value between 1 and 3",
+                self.tipo_documento_ref)
+
+        if self.ind_global and self.ind_global != 1:
+            raise ValueError(
+                "Only the value \"1\" is valid for the field 'ind_global'",
+                self.ind_global)
+
+        if (
+                len(self.folio_ref) < constants.DTE_REFERENCIA_FOLIO_MIN_LENGTH
+                or len(self.folio_ref) > constants.DTE_REFERENCIA_FOLIO_MAX_LENGTH
+        ):
+            raise ValueError(
+                "The length of 'folio_ref' must be a value between "
+                f"{constants.DTE_REFERENCIA_FOLIO_MIN_LENGTH} and "
+                f"{constants.DTE_REFERENCIA_FOLIO_MAX_LENGTH}",
+                self.folio_ref)
+
+        if (
+            self.fecha_ref < constants.DTE_REFERENCIA_FECHA_NOT_BEFORE
+            or self.fecha_ref > constants.DTE_REFERENCIA_FECHA_NOT_AFTER
+        ):
+            raise ValueError(
+                "The date 'fecha_ref' must be after "
+                f"{constants.DTE_REFERENCIA_FECHA_NOT_BEFORE} and before "
+                f"{constants.DTE_REFERENCIA_FECHA_NOT_AFTER}",
+                self.fecha_ref)
+
+        if self.razon_ref and len(self.razon_ref) > constants.DTE_REFERENCIA_RAZON_MAX_LENGTH:
+            raise ValueError(
+                "The maximum length allowed for `razon_ref` is "
+                f"{constants.DTE_REFERENCIA_RAZON_MAX_LENGTH}",
+                self.razon_ref)
+
+
+@pydantic.dataclasses.dataclass(
```

Comment:
I would move the conversion of `DteXmlData` to a Pydantic Dataclass into a separate commit and/or pull request.